Sources/SymbolKit/SymbolGraph/Symbol/OverloadData.swift

```diff
@@ -0,0 +1,37 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+extension SymbolGraph.Symbol {
+    /// A mixin to hold data about overloads of a symbol.
+    ///
+    /// This mixin is generated in ``SymbolGraph/createOverloadGroupSymbols()`` to hold information
+    /// about detected overloads of the symbol.
+    public struct OverloadData: Mixin {
```

Comment:
Just to confirm my understanding; this isn't intended to be Swift specific, right? In the future we'll want to expand support for overloads to other languages?

If not—if this is intended to be Swift specific—then I think we should add this extension under `SymbolGraph.Symbol.Swift` instead.

Reply:
Right. I wanted this to be generalizable to something like C++ in the future.=